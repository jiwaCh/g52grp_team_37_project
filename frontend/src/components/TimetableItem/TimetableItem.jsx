import React from 'react';
import proptypes from 'prop-types';
import { ListItem, ListItemAvatar, Avatar, ListItemText, makeStyles } from '@material-ui/core';
import cx from 'classnames';
import useStyles from './TimetableItem.styles';

/**
 * Timetable Item component
 *
 * Component for line itmes in the timetable component
 *
 * @param {{ busLine: string | number, time: string, destination: string, alternate: boolean, color: string}} props component props
 * @returns TimetableItem component
 */
const TimetableItem = ({ busLine, time, destination, alternate, color }) => {
  const classes = useStyles();
  const customClass = makeStyles(theme => ({
    lineColor: {
      color: theme.palette.getContrastText(color),
      backgroundColor: color,
    },
  }))();
  return (
    <ListItem dense={true} className={cx({ [classes.alternate]: alternate })}>
      <ListItemAvatar>
        <Avatar className={customClass.lineColor}>{busLine}</Avatar>
      </ListItemAvatar>
      <ListItemText primary={destination} secondary={time} />
    </ListItem>
  );
};

TimetableItem.propTypes = {
  busLine: proptypes.oneOfType([proptypes.string, proptypes.number]).isRequired,
  time: proptypes.string.isRequired,
  destination: proptypes.string.isRequired,
  alternate: proptypes.bool.isRequired,
  color: proptypes.string.isRequired,
};

export default TimetableItem;
