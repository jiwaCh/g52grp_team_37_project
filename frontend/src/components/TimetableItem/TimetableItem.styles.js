import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(theme => {
  const alternateColor = '#efefef';
  return {
    alternate: {
      color: theme.palette.getContrastText(alternateColor),
      backgroundColor: alternateColor,
    },
  };
});

export default useStyles;
