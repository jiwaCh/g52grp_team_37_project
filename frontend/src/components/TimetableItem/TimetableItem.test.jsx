import React from 'react';
import TimetableItem from './TimetableItem';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

it('renders correctly', () => {
  const props = {
    busLine: '31',
    time: '20:20',
    destination: 'Victora Centre',
    alternate: false,
    color: '#E37222',
  };

  const component = shallow(<TimetableItem {...props} />);

  expect(toJson(component)).toMatchSnapshot();
});
