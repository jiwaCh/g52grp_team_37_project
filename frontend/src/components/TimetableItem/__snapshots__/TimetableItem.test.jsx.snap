// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<WithStyles(ForwardRef(ListItem))
  className=""
  dense={true}
>
  <WithStyles(ForwardRef(ListItemAvatar))>
    <WithStyles(ForwardRef(Avatar))
      className="makeStyles-lineColor-2"
    >
      31
    </WithStyles(ForwardRef(Avatar))>
  </WithStyles(ForwardRef(ListItemAvatar))>
  <WithStyles(ForwardRef(ListItemText))
    primary="Victora Centre"
    secondary="20:20"
  />
</WithStyles(ForwardRef(ListItem))>
`;
